pkg/agent/plugin/workloadattestor/systemd/systemd_posix_test.go

```diff
@@ -0,0 +1,143 @@
+//go:build !windows
+// +build !windows
+
+package systemd
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/spire/pkg/agent/plugin/workloadattestor"
+	"github.com/spiffe/spire/test/plugintest"
+	"github.com/spiffe/spire/test/spiretest"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	ctx = context.Background()
+)
+
+func TestPlugin(t *testing.T) {
+	spiretest.Run(t, new(Suite))
+}
```

Comment:
This isn't something you should be expected to have known but we are not using testify Suite's in new code. They have poor UX and interactions with the go test tool.

Reply:
I can modify this, I wrote this in a style similar to the unix workloadattestor. I am fairly new to Go and have no clue what to replace it with. Can you point me to some existing test that does not use testify suite?

Reply:
> This isn't something you should be expected to have known but we are not using testify Suite's in new code. They have poor UX and interactions with the go test tool.

For the benefit of passerbys' education, what is the new preferred unit testing suite that everyone should be using?

Reply:
Let me clarify my position. `github.com/stretchr/testify/{require,assert}` are fantastic and used throughout the project. However, `github.com/stretchr/testify/suite` has not proved to be as useful and in many cases decreases maintainability, has a more difficult UX, and has friction with stdlib tooling.

In lieu of using the that specific Suite functionality, the codebase generally uses either [subtests](https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks) or table-driven tests when there is common setup/teardown that needs to happen for a series of tests.

Reply:
@azdagron Thanks for the feedback. I changed this to use subtests.